test/benchmark_csv_reporter.cc

```diff
@@ -0,0 +1,87 @@
+  public:
+	using StreamingReporterBase::StreamingReporterBase;
+
+	static std::string getDescription() { return "Reporter for benchmarks in CSV format"; }
+
+	void testRunStarting(Catch::TestRunInfo const& testRunInfo) override {
+		StreamingReporterBase::testRunStarting(testRunInfo);
+		fmt::print(m_stream, "test case,part,benchmark name,samples,iterations,estimated,mean,low mean,high mean,std dev,low std dev,high std dev\n");
+	}
+
+	void benchmarkStarting(Catch::BenchmarkInfo const& benchmarkInfo) override {
+		StreamingReporterBase::benchmarkStarting(benchmarkInfo);
+		fmt::print(m_stream, "{},{},{},{},{},{},", fmt::join(active_sections, " > "), active_part, escape_csv(benchmarkInfo.name), benchmarkInfo.samples,
+		    benchmarkInfo.iterations, benchmarkInfo.estimatedDuration);
+	}
+
+	void benchmarkEnded(Catch::BenchmarkStats<> const& benchmarkStats) override {
+		StreamingReporterBase::benchmarkEnded(benchmarkStats);
+		fmt::print(m_stream, "{:.4f},{:.4f},{:.4f},{:.4f},{:.4f},{:.4f}\n", benchmarkStats.mean.point.count(), benchmarkStats.mean.lower_bound.count(),
+		    benchmarkStats.mean.upper_bound.count(), benchmarkStats.standardDeviation.point.count(), benchmarkStats.standardDeviation.lower_bound.count(),
+		    benchmarkStats.standardDeviation.upper_bound.count());
+		// Flush so we can watch results come in when writing to file
+		m_stream.flush();
+	}
+
+	// It looks like benchmarkStarting is not called in case a benchmark fails, so we don't need to print the remaining columns either.
+	// TODO: Do we want to somehow report this anyway?
+	void benchmarkFailed(Catch::StringRef benchmarkName) override { StreamingReporterBase::benchmarkFailed(benchmarkName); }
+
+	void testCasePartialStarting(Catch::TestCaseInfo const& testInfo, uint64_t partNumber) override {
+		StreamingReporterBase::testCasePartialStarting(testInfo, partNumber);
+		active_part = partNumber;
+	}
+
+	void sectionStarting(Catch::SectionInfo const& sectionInfo) override {
+		StreamingReporterBase::sectionStarting(sectionInfo);
+		// Each test case has an implicit section with the name of the test case itself,
+		// so there is no need to capture that separately.
+		active_sections.push_back(escape_csv(sectionInfo.name));
+	}
+
+	void testCasePartialEnded(Catch::TestCaseStats const& testCaseStats, uint64_t partNumber) override {
+		StreamingReporterBase::testCasePartialEnded(testCaseStats, partNumber);
+		active_part = 0;
+		active_sections.clear();
+	}
+
+  private:
+	// It is difficult to distinguish test case invocations for different generator values, since
+	// as of Catch2 v.3.0.0-preview4 there is unfortunately no real metadata available (for example,
+	// it is not possible to get the current value returned for each of the generators).
+	// The only piece of information we have is the "part number", which we emit to distinguish
+	// different generator values. Note however that sections also increment the part number.
+	uint64_t active_part;
```

Comment:
The default text reporter also doesn't show generator values, does it? Including the part number is not a bad idea, but it's more an argument for not using generators in benchmarks.

Reply:
I agree that generators should better be avoided, but I also don't like emitting the same line twice in case somebody does use them anyway. I could try detecting generators being used and print a warning.

Reply:
I've implemented the warning.